### DISCUSSION TITLE
Custom perf instrumentation for [airflow -> node] flow without http header?

### DISCUSSION THREAD

**Discussion by pepf** [getsentry/sentry-python#1922] [Q&A] [answered]
We are in a situation where airflow is spinning up node.js Jobs dynamically on our kubernetes cluster, which in turn run "scripts" in our existing node.js codebase. We'd like to introduce distributed traces as a way to completely monitor this process e2e. both on airflow and on node.js level.

As far as I understand right now, it looks like sentry instrumentation would always add transaction ids to outgoing http requests. What I would like to work towards is a way to generate the transaction Id inside the airflow job, and whenever we create the "command" to run on the node.js instance we would be able to pass that transactionId along like so:
`/home/node/app/scripts/processStuff.js --SentryTransactionId=1234`
On the node.js end, we would then be able to "pick up" this transaction from the id? 
The creation/cleaning up of the job is also happening on airflow parts, so creating/finishing a given transaction seems to me would be feasable from the airflow part. Im mostly concerned about connecting the trace based on something else than a http header.

Is there any documentation on this? Is there any drawbacks that would make this difficult?
Thanks!

**Comment by antonpirker** [comment, marked as answer]
Hello @pepf !

You could create a transaction in airflow with `transaction = start_transaction(...)` and then you could give some information to the node script and in the node script do `transaction = Transaction.continue_from_headers(...)`. 

The information you need to give from airflow to the node script is the `baggage` and `sentry-trace` information that normally is sent via HTTP headers. 

We have discussions about supporting something like you suggested, but with environment variables here: https://github.com/getsentry/rfcs/blob/main/text/0071-continue-trace-over-process-boundaries.md

Unfortunately this RFC and the links in there are the only documentation we have for this.
Does this help you a bit?
